diff --git a/frameworks/innerkitsimpl/test/fuzztest/medialibraryrefreshmanager_fuzzer/BUILD.gn b/frameworks/innerkitsimpl/test/fuzztest/medialibraryrefreshmanager_fuzzer/BUILD.gn
index ecc051c7b99e10a5dc3a7fd46aeecafee3ca7fb0..eeb8aaff32f6cbbc75b59a330396b9c697db42ce 100644
--- a/frameworks/innerkitsimpl/test/fuzztest/medialibraryrefreshmanager_fuzzer/BUILD.gn
+++ b/frameworks/innerkitsimpl/test/fuzztest/medialibraryrefreshmanager_fuzzer/BUILD.gn
@@ -22,9 +22,13 @@ ohos_fuzztest("MediaLibraryRefreshManagerFuzzTest") {
     "./include",
     "${MEDIALIB_INNERKITS_PATH}/medialibrary_data_extension/include",
     "${MEDIALIB_SERVICES_PATH}/media_albums_refresh/include",
+    "${MEDIALIB_SERVICES_PATH}/media_cloud_sync_notify_handle/include",
   ]
 
-  sources = [ "src/medialibraryrefreshmanager.cpp" ]
+  sources = [ "src/medialibraryrefreshmanager.cpp",
+              "${MEDIALIB_SERVICES_PATH}/media_albums_refresh/src/albums_refresh_worker.cpp",
+              "${MEDIALIB_SERVICES_PATH}/media_albums_refresh/src/albums_refresh_notify.cpp",
+              "${MEDIALIB_SERVICES_PATH}/media_cloud_sync_notify_handle/src/cloud_album_handler.cpp"]
 
   deps = [
     "${MEDIALIB_INNERKITS_PATH}/media_library_helper:media_library",
diff --git a/frameworks/innerkitsimpl/test/fuzztest/medialibraryrefreshmanager_fuzzer/src/medialibraryrefreshmanager.cpp b/frameworks/innerkitsimpl/test/fuzztest/medialibraryrefreshmanager_fuzzer/src/medialibraryrefreshmanager.cpp
index 75ed71167442a5ff8e23a34cc2dc0cb2343f174a..deb2649b72048732eda1f3ae4b8039d269b9bae4 100644
--- a/frameworks/innerkitsimpl/test/fuzztest/medialibraryrefreshmanager_fuzzer/src/medialibraryrefreshmanager.cpp
+++ b/frameworks/innerkitsimpl/test/fuzztest/medialibraryrefreshmanager_fuzzer/src/medialibraryrefreshmanager.cpp
@@ -185,6 +185,7 @@ static void RefreshNotifyInfoTest(const uint8_t* data, size_t size)
     Media::CloudSyncNotifyInfo fuzzsyncnotifyinfo = FuzzCloudSyncNotifyInfo(data, size);
     std::vector<std::string> fuzzvector = FuzzVector(data, size);
     Media::AlbumsRefreshManager &instance = Media::AlbumsRefreshManager::GetInstance();
+    std::shared_ptr<Media::AlbumsRefreshWorker> refreshWorker = std::make_shared<Media::AlbumsRefreshWorker>();
     instance.RefreshPhotoAlbums(fuzznotifyinfo);
     instance.AddAlbumRefreshTask(fuzznotifyinfo);
     instance.NotifyPhotoAlbums(fuzznotifyinfo);
@@ -193,6 +194,10 @@ static void RefreshNotifyInfoTest(const uint8_t* data, size_t size)
     instance.CovertCloudId2AlbumId(g_rdbStore, fuzzvector);
     instance.CovertCloudId2FileId(g_rdbStore, fuzzvector);
     instance.RefreshPhotoAlbumsBySyncNotifyInfo(g_rdbStore, fuzznotifyinfo);
+    refreshWorker->StartConsumerThread();
+    refreshWorker->AddAlbumRefreshTask(fuzznotifyinfo);
+    refreshWorker->GetSystemAlbumIds(fuzznotifyinfo, fuzzvector);
+    refreshWorker->TryDeleteAlbum(fuzznotifyinfo, fuzzvector);
 }
 
 void SetTables()

